File: leetcode/src/main/java/com/giwankim/leetcode/UniquePaths.java
package com.giwankim.leetcode;

public class UniquePaths {
  public int uniquePaths(int m, int n) {
    int[][] a = new int[m + 1][n + 1];
    a[1][1] = 1;
    for (int i = 1; i <= m; i++) {
      for (int j = 1; j <= n; j++) {
        a[i][j] += a[i - 1][j] + a[i][j - 1];
      }
    }
    return a[m][n];
  }
}

File: leetcode-kt/src/main/kotlin/com/giwankim/leetcode/UniquePaths.kt
package com.giwankim.leetcode

class UniquePaths {
    fun uniquePaths(
        m: Int,
        n: Int,
    ): Int {
        val a = Array(m) { IntArray(n) { 1 } }
        for (i in 1 until m) {
            for (j in 1 until n) {
                a[i][j] = a[i - 1][j] + a[i][j - 1]
            }
        }
        return a[m - 1][n - 1]
    }
}
